src/code/z_skelanime.c

```diff
@@ -1,176 +1,1680 @@
+    func_800C6B54(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 0x37E);
+}
+
+#ifdef NON_MATCHING
+void SkelAnime_LodDrawLimbSV(GlobalContext* globalCtx, s32 limbIndex, u32* limbTable, Vec3s* actorDrawTable,
+                             SkelAnime_LimbUpdateMatrix updateMtxFunc, SkelAnime_LimbAppendDlist appendDlistFunc,
+                             Actor* actor, s32 dListIndex, Mtx** mtx) {
+    SkelLimbEntry* limbEntry;
+    Gfx* dList[2];
+    Vec3f pos;
+    Vec3s rot;
+    GraphicsContext* gfxCtx;
+    Gfx* gfx[3];
+
+    Matrix_Push();
+
+    limbEntry = SEGMENTED_TO_VIRTUAL(limbTable[limbIndex]);
+    limbIndex++;
+
+    rot = actorDrawTable[limbIndex];
+
+    pos.x = limbEntry->translation.x;
+    pos.y = limbEntry->translation.y;
+    pos.z = limbEntry->translation.z;
+    dList[1] = dList[0] = limbEntry->displayLists[dListIndex];
+
+    if ((updateMtxFunc == NULL) || (updateMtxFunc(globalCtx, limbIndex, &dList[1], &pos, &rot, actor) == 0)) {
+        Matrix_TranslateThenRotateZYX(&pos, &rot);
+        if (dList[1] != NULL) {
+            Matrix_ToMtx(*mtx, "../z_skelanime.c", 945);
+            gfxCtx = globalCtx->state.gfxCtx;
+            func_800C6AC4(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 946);
+            gSPMatrix(gfxCtx->polyOpa.p++, *mtx, G_MTX_LOAD);
+            gSPDisplayList(gfxCtx->polyOpa.p++, dList[1]);
+            func_800C6B54(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 949);
+            (*mtx)++;
+        } else if (dList[0] != 0) {
+            Matrix_ToMtx(*mtx, "../z_skelanime.c", 954);
+            (*mtx)++;
+        }
+    }
+    if (appendDlistFunc != NULL) {
+        appendDlistFunc(globalCtx, limbIndex, &dList[0], &rot, actor);
+    }
+    if (limbEntry->firstChildIndex != LIMB_DONE) {
+        SkelAnime_LodDrawLimbSV(globalCtx, limbEntry->firstChildIndex, limbTable, actorDrawTable, updateMtxFunc,
+                                appendDlistFunc, actor, dListIndex, mtx);
+    }
+
+    Matrix_Pull();
+
+    if (limbEntry->nextLimbIndex != LIMB_DONE) {
+        SkelAnime_LodDrawLimbSV(globalCtx, limbEntry->nextLimbIndex, limbTable, actorDrawTable, updateMtxFunc,
+                                appendDlistFunc, actor, dListIndex, mtx);
+    }
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/SkelAnime_LodDrawLimbSV.s")
+#endif
+
+#ifdef NON_MATCHING
+void SkelAnime_LodDrawSV(GlobalContext* globalCtx, u32* limbTable, Vec3s* actorDrawTable, s32 limbCount,
+                         SkelAnime_LimbUpdateMatrix updateMtxFunc, SkelAnime_LimbAppendDlist appendDlistFunc,
+                         Actor* actor, s32 dListIndex) {
+    SkelLimbEntry* limbEntry;
+    Gfx* dList[2];
+    Vec3f pos;
+    Vec3s rot;
+    Mtx* mtx;
+    GraphicsContext* gfxCtx;
+    Gfx* gfx[4];
+
+    do {
+        if (1) {
+            mtx = Graph_Alloc(globalCtx->state.gfxCtx, limbCount * 0x40);
+        }
+    } while (0);
+
+    if (limbTable == NULL) {
+        osSyncPrintf("\x1b[31m");
```

Comment:
these weird looking characters are used to set the color of the debug output. check out macros in vt.h and try to apply them anywhere you see it

Reply:
Fixed in f2c65d6